### PR TITLE
Allow enabling and disabling vertical autoscaling

### PR DESCRIPTION
**Pull request by aguilinger**
Creates the most basic functionality for vertical autoscaling: enable and disable. It also only conditionally shows the autoscaling section if the stack is enabled to show it (though it really isn't a stack configuration, the variable just happened to exist there already).

Stack with `vertical_autoscaling: false`
Screenshot 2023-11-08 at 4 04 49 PM

Same stack with `vertical_autoscaling: true`
Screenshot 2023-11-08 at 4 04 19 PM

### CODE REVIEW

**Comment by aguilinger** [comment]
Note: must wait on https://github.com/aptible/deploy-api/pull/1282 before merge

**Comment by joeclayallday** [comment]
Nice work! Looks good

File: src/ui/pages/app-detail-service-scale.tsx [hunk 1]
```diff
@@ -152,170 +263,179 @@ export const AppDetailServiceScalePage = () => {
...
-              {service.containerCount} container
-              {service.containerCount > 1 ? "s" : ""} x{" "}
-              {service.containerMemoryLimitMb / 1024} GB x $
-              {currentPricePerHour} per GB/hour
-            </p>
-          </div>
-          <div>
-            <p className="text-black-500">Current Estimated Monthly Cost</p>
-            <p className="text-right text-lg text-green-400">${currentPrice}</p>
+    <div className="flex flex-col gap-2">
```
Comment (joeclayallday) [inline src/ui/pages/app-detail-service-scale.tsx:266]: @aguilinger Tiny change here, on this line change `gap-2` to `gap-4` to be consistent with other spacing between boxes. Here's a spacing example on the App / Settings Tab for visual comparison:
Screenshot 2023-11-08 at 7 30 14 PM

Comment (aguilinger) [reply src/ui/pages/app-detail-service-scale.tsx]: Thanks! Here's gap-4
Screenshot 2023-11-09 at 8 39 43 AM

File: src/ui/pages/app-detail-service-scale.tsx [hunk 2]
```diff
@@ -41,6 +46,109 @@ type AppScaleProps = {
...
+        post_scale_up_cooldown_seconds: number;
+        post_scale_down_cooldown_seconds: number;
+        post_release_cooldown_seconds: number;
+        mem_cpu_ratio_r_threshold: number;
+        mem_cpu_ratio_c_threshold: number;
+        mem_scale_up_threshold: number;
+        mem_scale_down_threshold: number;
+        minimum_memory: number;
+      }[];
+    }>
```
Comment (neurosnap) [inline src/ui/pages/app-detail-service-scale.tsx:71]: Please move this into a named type above the react component.

File: src/ui/pages/app-detail-service-scale.tsx [hunk 3]
```diff
@@ -152,170 +263,179 @@ export const AppDetailServiceScalePage = () => {
...
-    </Box>
+
+            {changesExist ? (
+              <Button
+                className="w-40 ml-2 flex font-semibold"
+                onClick={() => {
+                  setContainerProfileType(service.instanceClass);
+                  setContainerSize(service.containerMemoryLimitMb);
+                  setContainerCount(service.containerCount);
+                }}
```
Comment (neurosnap) [inline src/ui/pages/app-detail-service-scale.tsx:430]: Please move this into a named const function inside the react function body.

File: src/ui/pages/app-detail-service-scale.tsx [hunk 4]
```diff
@@ -41,6 +46,109 @@ type AppScaleProps = {
...
+        mem_cpu_ratio_c_threshold: number;
+        mem_scale_up_threshold: number;
+        mem_scale_down_threshold: number;
+        minimum_memory: number;
+      }[];
+    }>
+  >(fetchServiceSizingPoliciesByServiceId({ id }));
+  const dispatch = useDispatch();
+
+  const isScaling: boolean =
```
Comment (neurosnap) [inline src/ui/pages/app-detail-service-scale.tsx:75]: We could probably simplify this a little bit by using optional chaining.  Non-blocking.

**Review by neurosnap — COMMENTED** [review]
Looks pretty good to me!

**Review by joeclayallday — APPROVED** [review]

**Review by neurosnap — APPROVED** [review]

**Pull request merged**